utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,298 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1 and assign the value of the upper magnitude to
+    /// the magnitude.
+    fn increment_abs_by_one(&mut self) {
+        let is_digits_empty = self.digits.is_empty();
+
+        for i in (0..self.digits.len()).rev() {
+            self.digits[i] += 1;
+            if self.digits[i] < 10 {
+                #[cfg(debug_assertions)]
+                self.check_invariants();
+                return;
+            }
+
+            self.digits.pop();
+        }
+
+        if self.magnitude == i16::MAX {
+            self.magnitude = 0;
+
+            #[cfg(debug_assertions)]
+            self.check_invariants();
+            return;
+        }
+
+        // Still a carry, carry one to the next magnitude.
+        self.digits.push(1);
+        self.magnitude = {
```

Comment:
if the `digits` was empty, then we add one to the digits and the magnitude will equal to the `upper_magnitude`

Reply:
We should not be relying on `upper_magnitude` to give us this information. That only purpose of that variable is to set the leading zeros on the FixedDecimal.

If `is_digits_empty` is true, we should forbid this function from being called:

```rust
debug_assert!(!self.digits.is_empty());
```

and handle this case at the call site instead.

Reply:
Ack. I have changed the logic.